src/client/gui.js

```diff
@@ -1,8 +1,236 @@
-/* globals ProjectDialogMorph, ensureFullUrl, Morph, AlignmentMorph, InputFieldMorph, localize,
+/* globals ProjectDialogMorph, world, SymbolMorph, PushButtonMorph, ToggleButtonMorph, ensureFullUrl, Morph, AlignmentMorph, InputFieldMorph, localize,
    Point, TextMorph, Color, nop, ListMorph, IDE_Morph, Process, BlockImportDialogMorph,
    BlockExportDialogMorph, detect, SnapCloud, SnapSerializer, ScrollFrameMorph,
    DialogBoxMorph, SnapActions, SpeechBubbleMorph
    */
+
+function isMobileDevice() {
+    return (typeof window.orientation !== 'undefined') || (navigator.userAgent.indexOf('Mobile') !== -1);
+}
+
+var _toggleAppMode = IDE_Morph.prototype.toggleAppMode;
+IDE_Morph.prototype.toggleAppMode = function() {
+    var ideMorphIns = world.children[0];
+    _toggleAppMode.apply(ideMorphIns, arguments);
+    if (isMobileDevice()) {
+        mobileMode.init();
+    }
+};
+
+var _ideFixLayout = IDE_Morph.prototype.fixLayout;
+IDE_Morph.prototype.fixLayout = function() {
+    var ideMorphIns = world.children[0];
+    _ideFixLayout.apply(ideMorphIns, arguments);
+    if (ideMorphIns.isAppMode &&  isMobileDevice()) {
+        // if mobilemode is fully initialized
+        if (mobileMode.buttons.length !== 0) mobileMode.fixLayout();
+    }
+};
+
+var mobileMode = {
+    // assuming top pos for horizontal alignment and right for vertical
+    _stackMode: '', // stack controls horizontally or vertically
```

Comment:
This might be better as a flag rather than a string... something like `isVertical` or `isVerticalStack`

Reply:
as-is I'm using this variable for 3 different states so it wouldn't cleanly translate into a boolean. However, I could use undefined as a state an have the values be `true` or `false`